swiggyclone/service/views.py

```diff
@@ -89,3 +88,57 @@ def post(self,request):
             return Response(serializer.data, status=status.HTTP_201_CREATED)
         return Response(serializer.error_messages,
                         status=status.HTTP_400_BAD_REQUEST)
+
+
+class RestaurentViewSet(viewsets.ModelViewSet):
+    queryset = Restaurent.objects.all().prefetch_related('branches')
+    serializer_class = RestaurentSerializers
+
+
+class BranchViewSet(viewsets.ModelViewSet):
+    queryset = Branch.objects.all().prefetch_related('foodItems')
+    serializer_class = BranchSerializer
+    def get_queryset(self):
+        return Branch.objects.select_related('foodItem').filter(restaurent_id = self.kwargs['restaurent_pk'])
+
+    @action(detail=True, method=['POST'])
+    def placeOrder(self, request, pk):
+        branch = self.get_object()
+        foodItems = FoodItem().objects.filter(branch=branch)
+        foodItemsSerializer = FoodItemSerializer(foodItems, many=True)
+        orderSerializers = OrderSerializers(data=request.data)
+        data = orderSerializers.data
+        orderDiscriptions = data['orderDiscriptions']
+        branch = Branch.objects.select_related('foodItem').get(id=pk)
+        for orderDiscription in orderDiscriptions:
+            orderFoodItem = orderDiscription['foodItem']
+            orderQuantity = orderDiscription['quantity']
+            try:
+                instance = FoodItem.filter(branch_id=orderedBranch, id=orderFoodItem)
+            except FoodItem.DoesNotExist as e:
+                return Response({'error':'Given fooditem not found'},status = 404)
+            if instance[0].quantity < orderQuantity:
+                return Response({'error':'Given fooditem not found'},status = 404)
+        total_price = 0
+        for orderDiscription in orderDiscriptions:
+            orderFoodItem = orderDiscription['foodItem']
+            orderQuantity = orderDiscription['quantity']
+            instance = FoodItem.filter(branch_id=orderedBranch, id=orderFoodItem)
+            foodItemPrice = instance.first().price
+            finalQuantity = instance.first().quantity - orderQuantity
+            serializer = FoodItemSerializer(instance[0], data = {"quantity":finalQuantity}, partial = True)
```

Comment:
why instance[0]? why not instance.first()?

Reply:
Okk